vscode/src/qirGeneration.ts

```diff
@@ -25,15 +25,28 @@ export async function getQirForActiveWindow(): Promise<string> {
     );
   }
 
+  const configuration = vscode.workspace.getConfiguration("Q#");
   // Check that the current target is base profile, and current doc has no errors.
-  const targetProfile = vscode.workspace
-    .getConfiguration("Q#")
-    .get<string>("targetProfile", "full");
+  const targetProfile = configuration.get<string>("targetProfile", "full");
   if (targetProfile !== "base") {
-    throw new QirGenerationError(
-      "Submitting to Azure is only supported when targeting the QIR base profile. " +
-        "Please update the QIR target via the status bar selector or extension settings."
+    const result = await vscode.window.showWarningMessage(
+      "Submitting to Azure is only supported when targeting the QIR base profile.",
+      { modal: true },
+      { title: "Change the QIR target profile and continue", action: "set" },
+      { title: "Cancel", action: "cancel", isCloseAffordance: true }
     );
+    if (result?.action !== "set") {
+      throw new QirGenerationError(
+        "Submitting to Azure is only supported when targeting the QIR base profile. " +
+          "Please update the QIR target via the status bar selector or extension settings."
+      );
+    } else {
+      await configuration.update(
```

Comment:
Are we sure this change to configuration will be done, the language service updated, the documents revalidated for base profile, and the diagnostics up to date for Base profile before this 'await' returns and it moves on to check the diagnostics are empty in the following code?

Reply:
Nope, not at all sure :/ I found it very tricky to guarantee this, so I just verified manually that the events seem to get fired in the correct order and we move on. But I can't be sure it will always work. It would be fairly complex to wire up the language service to force a diagnostics "refresh" at this point. 

Worst case, we set the base profile, but still sometimes fail out of this function due to compilation errors further down this function. In that case, the user can retry.

I don't love this, so if I missed a simpler solution please let me know.

Reply:
Relying on editor squigglies to validate the document was kind of a race-y approach to begin with. This works today since our language service is synchronous (it's not even running in a separate worker), but as projects grow, you can imagine a TypeScript-like approach where we batch/delay computing diagnostics for performance reasons. So you can't be 100% sure that the diagnostics are perfectly in sync with the document contents.

Reply:
We could add a `setTimeout`?... I kid, I kid. OK, let's go with it for now, and figure out any 'cleaner' approach later.

I did initially want to call 'compile' directly and see if there were any errors, but as we removed that from the compiler and it's only on the language service now 🙄 , and I didn't want to create a new language service, this "checking if the doc has errors currently" was kind of a hack anyway.

Reply:
Well, at the end of the day, `getQir` will fail if there are compilation errors. I think that's all we need. You shouldn't need to call the compiler twice. Maybe we could refine the error messaging, but the behavior is correct. 

I do _like_ the squiggly check, as race-y as it is. It's a nice, best-effort check to fail fast in case something looks suspicious. But in general, the language service should just be there to "decorate" the editor, not to interact with codegen or the runtime.